crates/librqbit_core/src/magnet.rs

```diff
@@ -54,11 +77,90 @@ impl std::fmt::Display for Magnet {
     }
 }
 
+/// A parsed magnet link, supporting both v1 and v2 links.
+pub struct MagnetBEP52 {
```

Comment:
i'd probably go a bit further, and change the existing "Magnet" struct to contain "InfoHash" instead of "Id20". Whatever expects only Id20 would say "I only support BTV1 info hashes" or smth like that.

From memory (not looking at the code), the only place that works with magnet links is session, so you can just stick the match in there, or even better, make a ```fn as_id20(&self) -> Option<Id20>``` method on InfoHash, and then in e.g session add ```.context("magnet link doesn't contain BTv1 info hash")```